### PR TITLE
Mar 1051 - patch

### PR DESCRIPTION
**Pull request by jamespacileo**

### CODE REVIEW

File: api/barriers/models.py [hunk 1]
```diff
@@ -1163,12 +1163,13 @@ def priority_filter(self, queryset, name, value):
...
     def has_action_plan_filter(self, queryset, name, value):
-        if value:
-            from api.action_plans.models import ActionPlanMilestone
+        if not value:
+            return queryset
 
-            milestones = ActionPlanMilestone.objects.all()
-            return queryset.filter(action_plans__milestones__in=milestones)
-        return queryset
+        from api.action_plans.models import ActionPlanMilestone
```
Comment (kevincarrogan) [inline api/barriers/models.py:1169]: Why is this import happening here?

File: api/barriers/models.py [hunk 2]
```diff
@@ -1163,12 +1163,13 @@ def priority_filter(self, queryset, name, value):
...
+
+        milestones = ActionPlanMilestone.objects.all()
+        return queryset.filter(action_plans__milestones__in=milestones)
```
Comment (kevincarrogan) [inline api/barriers/models.py:1172]: This kind of confuses me. What's the actual intention here?

Comment (jamespacileo) [reply api/barriers/models.py:1172]: it's just swapping the flow so the main logic isn't nested in an if

File: api/barriers/models.py (hunk 1 above)
Comment (jamespacileo) [reply api/barriers/models.py:1169]: import causes a circular dependency, possibly we could move this whole class outside into a third file and that would resolve this

we can either refactor this now or add a TODO comment and a backlog ticket to solve this when we have more time

**Review by kevincarrogan — APPROVED** [review]

File: api/barriers/models.py (hunk 2 above)
Comment (jamespacileo) [reply api/barriers/models.py:1172]: yes it's returning the list of items that have at least 1 milestone on the action plan model

**Pull request merged**